bloptools/gp/__init__.py

```diff
@@ -12,208 +11,99 @@
-        )
-
-    def forward(self, x):
-        return gpytorch.distributions.MultivariateNormal(self.mean_module(x), self.covar_module(x))
-
-
-class GPR:
-    """
-    A Gaussian process regressor, with learning methods.
-    """
-
-    def __init__(self, length_scale_bounds=(1e-3, 1e0), max_noise_fraction=1e-2):
-        self.max_noise_fraction = max_noise_fraction
-        self.state_dict = None
-        self.length_scale_bounds = length_scale_bounds
-
-    def set_data(self, x, y):
-        """
-        Set the data with parameters and values.
-
-        x: parameters
-        y: function values at those parameters
-        """
-
-        if np.isnan(y).any():
-            raise ValueError("One of the passed values is NaN.")
-
-        self.x, self.y = np.atleast_2d(x), np.atleast_1d(y)
-        self.n, self.n_dof = self.x.shape
-
-        # prepare Gaussian process ingredients for the regressor and classifier
-        # use only regressable points for the regressor
-        self.inputs = torch.as_tensor(self.x).float()
-        self.targets = torch.as_tensor(self.y).float()
-
-        self.noise_upper_bound = 1e-1 * self.max_noise_fraction if len(self.y) > 1 else self.max_noise_fraction
-        self.likelihood = gpytorch.likelihoods.GaussianLikelihood(
-            noise_constraint=gpytorch.constraints.Interval(0, self.noise_upper_bound)
-        )
-
-        self.model = GaussianProcessModel(
-            self.inputs,
-            self.targets,
-            self.likelihood,
-            self.n_dof,
-            self.length_scale_bounds,
-        )
-
-        self.init_state_dict = self.model.state_dict()
-        if self.state_dict is not None:
-            self.model.load_state_dict(self.state_dict)
-
-    def train(self, training_iter=100, reuse_hypers=True, verbose=True):
-        if not reuse_hypers:
-            self.model.load_state_dict(self.init_state_dict)
-
-        self.likelihood.train()
-        self.model.train()
-
-        # Use the adam optimizer
-        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=1e-1)
-
-        # "Loss" for GPs - the marginal log likelihood
-        self.mll = gpytorch.mlls.ExactMarginalLogLikelihood(self.likelihood, self.model)
-
-        for i in range(training_iter):
-            self.optimizer.zero_grad()
-            loss = -self.mll(self.model(self.inputs), self.targets)
-            loss.backward()
-            self.optimizer.step()
-
-            if verbose and ((i + 1) % 100 == 0):
-                print(f"{i+1}/{training_iter} inverse_length_scales: {self.model.covar_module.trans_diagonal}")
-
-        self.state_dict = self.model.state_dict()
-
-    def regress(self, x):
-        x = torch.as_tensor(np.atleast_2d(x)).float()
+def load(filepath, **kwargs):
+    with h5py.File(filepath, "r") as f:
+        params = f["params"][:]
+    data = pd.read_hdf(filepath, key="data")
+    return Optimizer(init_params=params, init_data=data, **kwargs)
```

Comment:
Where do we get `Optimizer` from? Should it be imported? Or is it defined in this module?

Reply:
@thomaswmorris, could you please add a unit test for this load function?